src/main/java/com/mariamkatamashvili/trainerwork/messaging/MessageConsumer.java

```diff
@@ -0,0 +1,41 @@
+package com.mariamkatamashvili.trainerwork.messaging;
+
+import com.mariamkatamashvili.trainerwork.dto.WorkloadDTO;
+import com.mariamkatamashvili.trainerwork.exception.WorkloadException;
+import com.mariamkatamashvili.trainerwork.security.JwtTokenProvider;
+import com.mariamkatamashvili.trainerwork.service.WorkloadService;
+import jakarta.jms.Message;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class MessageConsumer {
+    private final WorkloadService workloadService;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    @JmsListener(destination = "${messaging.jms.destination.main}")
+    public void receiveWorkload(WorkloadDTO workload, Message message) {
+        extractAndValidateToken(workload, message);
+    }
+
+    public void extractAndValidateToken(WorkloadDTO workload, Message message) {
+        try {
+            String token = message.getStringProperty("Authorization");
+            if (token != null && token.startsWith("Bearer ")) {
+                token = token.substring(7);
+                if (jwtTokenProvider.isTokenValid(token)) {
```

Comment:
This verification would be better to expose into a separate component like AuthValidator or similar. Juts to call a separeate method like authValidator.validate and do all necesary check and produce exception inside. That will look much more clear.